doc/hardware/porting/board_porting.rst

```diff
@@ -158,6 +162,10 @@ The mandatory files are:
    software configuration in :ref:`kconfig` formats. This provides default
    settings for software features and peripheral drivers.
 
+.. note::
+
+	For **downstream** boards, the only mandatory file is the _defconfig file. This is because downstream boards only really need to use Kconfig, while upstream boards are required to be documented and support devicetree, twister, and Kconfig. If this file does not exist, then the build system will assume the user has selected an invalid board.
+
```

Comment:
> For **downstream** boards, the only mandatory file is the _defconfig file. This is because downstream boards only really need to use Kconfig, while upstream boards are required to be documented and support devicetree, twister, and Kconfig.

Those files are already described as `The mandatory files are:` so why also a `note` ?

Also, in the board porting guide we should not describe that devicetree file is optional, cause that is considered very advanced usage.

Describing dts this way can lead some users into confusion and other things will not work as expected if they start to not create the dts file.

> If this file does not exist, then the build system will assume the user has selected an invalid board.

this is really an implementation detail of how the build system work.
If anything should be done in this regard, then it will probably be better to improve the error message and not just print valid boards.

Reply:
Okay mainly I just wanted to point out the difference for downstream boards, I will fix this